lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,471 @@
+      end
+
+      def authorize(amount, credit_card, options = {})
+        #   byebug
+        params = {}
+        action = 'purchase'
+        authOnly = true
+        isSettleFunds = false
+        add_bank_amount_purchase(params, amount, authOnly)
+        add_credit_card(params, credit_card, action)
+        add_type_merchant_purchase(params, options[:merchant], isSettleFunds, options)
+        commit(action, params, '', '', options)
+      end
+
+      def refund(amount, credit_card, authCode, options, key_secret)
+        #   byebug
+        params = {}
+        action = 'refund'
+        add_bank_amount_refund(params, amount, authCode)
+        add_credit_card(params, credit_card, action)
+        add_type_merchant_refund(params, options)
+        commit(action, params, '', '', key_secret)
+      end
+
+      def capture(amount, authorization, authCode, options = {})
+        #   byebug
+        params = {}
+        action = 'capture'
+        add_capture(params, amount, authorization, authCode, options)
+        commit(action, params, '', '', options)
+      end
+
+      def add_capture(params, amount, authorization, authCode, options)
+        # byebug
+        params['amount'] = amount
+        params['authCode'] = authCode
+        params['merchantId'] = options[:merchant]
+        params['paymentToken'] = authorization
+        params['shouldGetCreditCardLevel'] = true
+        params['source'] = 'Spreedly'
+        params['tenderType'] = 'Card'
+      end
+
+      def void(iid, options)
+        # byebug
+        action = 'void'
+        commit(action, '', iid, '', options)
+      end
+
+      def verify(creditcardnumber)
+        # byebug
+        #   test = request_verify_headers
+        action = 'verify'
+        commit(action, '', '', creditcardnumber, '')
+      end
+
+      def getpaymentstatus(batchid, options)
+        action = 'getpaymentstatus'
+        commit(action, batchid, '', '', options)
+      end
+
+      def closebatch(batchid, options)
+        action = 'closebatch'
+        commit(action, batchid, '', '', options)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
+          gsub(%r((number)\W+\d+), '\1[FILTERED]').
+          gsub(%r((cvv)\W+\d+), '\1[FILTERED]')
+      end
+      # private
+
+      def add_customer_data(post, options); end
+
+      def add_address(post, creditcard); end
+
+      def add_bank_amount_purchase(params, amount, authOnly)
+        params['achIndicator'] = nil
+        params['amount'] = amount
+        params['authCode'] = nil
+        params['authOnly'] = authOnly
+        params['bankAccount'] = nil
+      end
+
+      def add_bank_amount_refund(params, amount, authCode)
+        params['amount'] = amount
+        params['authCode'] = authCode
+        params['authMessage'] = 'Approved or completed successfully.'
```

Comment:
Should this auth message be something we set? This looks like a message you would expect to get back from the server.